### PR TITLE
[wallet-ext] open Ledger connection modal in new tab to prevent HID/USB connection issues

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 
TLDR; Some browser APIs don't work properly when an extension is in popup mode, so we have to re-work some of the Ledger flow UX. The first UX change is that we need to always open the Ledger connection/import flow in a new window, which is what this PR changes. The second UX change (will tackle this in a follow-up) is that we need to stop requesting user connections during the signing flow which is something that I introduced yesterday.

https://capture.dropbox.com/ArbF7mmK4Ws1OkYj

### More information re: the root bug
`@ledger/transport-web-hid` relies on the `HID.requestDevices` API implemented in Chrome, and you can only call this API in the context of a user gesture (mousedown, keydown, etc.). The API uses this logic (https://developer.mozilla.org/en-US/docs/Web/API/UserActivation) to determine whether or not the API call fits in the context of a user gesture, but the logic simply doesn't work as expected when the Chrome extension is in popup mode.

## Test Plan 
- Manual testing
  - Can connect to Ledger successfully 100% of the time now
  - Despite demo'ing this several times, I can't seem to connect to Ledger in popup mode whatsoever now

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/components/ledger/ConnectLedgerModal.tsx [hunk 1]
```diff
@@ -10,14 +10,12 @@ import { ModalDialog } from '_src/ui/app/shared/ModalDialog';
 import { Text } from '_src/ui/app/shared/text';
 
 type ConnectLedgerModalProps = {
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/components/ledger/ConnectLedgerModal.tsx:12]: The modal is now accessible via a nested route, so there's no need to control the modal's visibility with a prop

File: apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            onConfirm={() => {
+                navigate(importLedgerAccountsUrl);
+            }}
+        />
+    );
+}
+
+function getLedgerErrorMessage(error: unknown) {
+    if (error instanceof LockedDeviceError) {
+        return 'Your device is locked. Un-lock it and try again.';
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx:38]: `Un-lock` -> `Unlock` 😛

**Review by pchrysochoidis — APPROVED** [review]

**Review by mamos-mysten — APPROVED** [review]

File: apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx (hunk 2 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx:38]: thanks haha, this was copy-pasted over and I've been meaning to consolidate all of the error handling/messaging 😛

File: apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                toast.error(getLedgerErrorMessage(error));
+            }}
+            onConfirm={() => {
+                navigate(importLedgerAccountsUrl);
+            }}
+        />
+    );
+}
+
+function getLedgerErrorMessage(error: unknown) {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx:36]: Aren't these our errors? Can we just make the `message` on it human readable?

**Review by Jordan-Mysten — APPROVED** [review]

File: apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx (hunk 3 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx:36]: I was going to re-work this in https://mysten.atlassian.net/browse/APPS-642.

Some of the errors are from the Ledger library and some are errors that we've intercepted and converted to our own errors. Right now, we're swallowing the internal exceptions which aren't good and hard for debugging so this was one thing I was going to address. I guess to directly answer your question, I didn't want to couple internal error messages to user messaging; I was thinking the errors we surface would have messages like:

```
LedgerConnectionFailedError(`Failed to connect to the user's Ledger device: ${internalErrorMessageFromLedger}`)
UI message -> Ledger connection failed. Try again.
```

Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx:39]: Going to go ahead and land but feel free to reply on this one :)
